examples/smart_contracts/v1/javascript/AccountDelegation.js

```diff
@@ -1,76 +1,56 @@
-        if (pendingInfo.round !== null && pendingInfo.round > 0) {
-            //Got the completed Transaction
-            console.log("Transaction " + pendingInfo.tx + " confirmed in round " + pendingInfo.round);
-            break;
-        }
-        lastround++;
-        await algodclient.statusAfterBlock(lastround);
-    }
-};
-// create an algod client
-let algodclient = new algosdk.Algod(token, server, port);
-(async () => {
+// create v2 client
+const algodClient = new algosdk.Algodv2(token, server, port);
 
+const main = async () => {
     // get suggested parameters
-    let params = await algodclient.getTransactionParams();
-    let endRound = params.lastRound + parseInt(1000);
-    let fee = await algodclient.suggestedFee();
-
-    // create logic sig
-    // b64 example "ASABACI=" is `int 0`
-    // see more info here: https://developer.algorand.org/docs/features/asc1/sdks/#accessing-teal-program-from-sdks
-    // let program = new Uint8Array(Buffer.from("base64-encoded-program" < PLACEHOLDER >, "base64"));
-    let program = new Uint8Array(Buffer.from("ASABACI=" , "base64"));
-    let lsig = algosdk.makeLogicSig(program);
+    const params = await algodClient.getTransactionParams().do();
+    params.fee = 1000;
+    params.flatFee = true;
+
+    const filePath = path.join(__dirname, '/teal/samplearg.teal');
+    const data = fs.readFileSync(filePath);
+    const  results = await algodClient.compile(data).do();
+    const program = new Uint8Array(Buffer.from(results.result, "base64"));
+
+    const args = [];
+    args.push(algosdk.encodeUint64(123));
+   
+    const lsig = algosdk.makeLogicSig(program, args); 
+    
+    // *** Begin account delegation changes ***
+    // import your private key mnemonic
+    const PASSPHRASE = "< mnemonic >";
+    const myAccount = algosdk.mnemonicToSecretKey(PASSPHRASE);
 
     // sign the logic signature with an account sk
     lsig.sign(myAccount.sk);
-
-    // create a transaction
-    let txn = {
-        "from": myAccount.addr,
-        // "to": "receiver-address" < PLACEHOLDER >,
-        "to": "SOEI4UA72A7ZL5P25GNISSVWW724YABSGZ7GHW5ERV4QKK2XSXLXGXPG5Y",
-        "fee": params.fee,
-        "amount": 10000,
-        "firstRound": params.lastRound,
-        "lastRound": endRound,
-        "genesisID": params.genesisID,
-        "genesisHash": params.genesishashb64
-    };
-
-    // Create the LogicSigTransaction with contract account LogicSig 
-    let rawSignedTxn = algosdk.signLogicSigTransaction(txn, lsig);
-
-    // send raw LogicSigTransaction to network  
-    console.log("This is expected to fail as the program is int 0 and will return false : ");      
-    let tx = (await algodclient.sendRawTransaction(rawSignedTxn.blob));
+    // *** End account delegation changes ***
+
+    // Setup a transaction
+    const sender = myAccount.addr;
+    const receiver = "GVVIH6IE3ZAFLIQF6UFBOH4TI5ZOPHFRMDQUTXQRKZ6XWSRTVRDBQ4GZLY";
```

Comment:
Please change to return funds to the "dispenser v2"

HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA